### ISSUE TITLE
PartialObservationWrapper's PERIODIC mode accepts wrong-shaped masks and silently corrupts observation shape

### ISSUE THREAD

**Issue by Svector-anu** [SlopDotCash/asi#147]
## Problem

`PartialObservationWrapper`'s `PERIODIC` mode stacks schedule masks first and only checks the stacked array's second axis (`sched.shape[1] == feature_dim`). a mask shaped `(feature_dim, 1)` passes that check spuriously, since stacking N such masks along axis 0 gives `(N, feature_dim, 1)`, whose `shape[1]` equals `feature_dim` by coincidence of the wrong axis being checked.

the malformed mask then broadcasts against a `(feature_dim,)` observation in `jnp.where`, silently turning it into `(feature_dim, feature_dim)` instead of raising.

## Reproduction (current main)

```text
>>> schedule = [jnp.array([True, False, True]).reshape(3,1) for _ in range(2)]
>>> sched = jnp.stack(schedule, axis=0)
>>> sched.shape
(2, 3, 1)
>>> sched.shape[1]  # the old check compares this to feature_dim=3, passes incorrectly
3

>>> obs = jnp.array([1.0, 2.0, 3.0])
>>> jnp.where(sched[0], obs, jnp.zeros_like(obs)).shape
(3, 3)   # documented shape is (3,), a feature vector silently became a matrix
```

## Reachability

the class is exported from the top-level public package. repo search found no in-tree production constructor call beyond the implementation's own docstring example and tests, but it's public API, any user reaches this with an ordinary array-like schedule argument, no hardcoded-safe input prevents it.

## Proposed fix

validate each mask's exact shape against `(feature_dim,)` before stacking, so a malformed mask is rejected at construction instead of silently corrupting the observation shape at step time.
